packages/jest-config/src/normalize.ts

```diff
@@ -46,6 +47,39 @@ type AllOptions = Config.ProjectConfig & Config.GlobalConfig;
 const createConfigError = (message: string) =>
   new ValidationError(ERROR, message, DOCUMENTATION_NOTE);
 
+function verifyDirectoryExists(path: Config.Path, key: string) {
+  try {
+    const rootStat = statSync(path);
+
+    if (!rootStat.isDirectory()) {
+      throw createConfigError(
+        `  ${chalk.bold(path)} in the ${chalk.bold(
+          key,
+        )} option is not a directory.`,
+      );
+    }
+  } catch (err) {
+    if (err instanceof ValidationError) {
+      throw err;
```

Comment:
printing a warning instead of these 3 `throw`s is trivial if we don't want a breaking change